integration/benchmarks_test.go

```diff
@@ -178,3 +182,129 @@ func BenchmarkInsertMany(b *testing.B) {
 		}
 	}
 }
+
+func BenchmarkInsertManyIntoDifferentCollections(b *testing.B) {
+	ctx, collection := setup.Setup(b)
+
+	provider := shareddata.BenchmarkSmallDocuments
+	iter := provider.NewIterator()
+
+	insertDocs := []any{}
+
+	for {
+		docs, err := iterator.ConsumeValues(iter)
+		if errors.Is(err, iterator.ErrIteratorDone) || docs == nil {
+			break
+		}
+
+		require.NoError(b, err)
+
+		for _, doc := range docs {
+			insertDocs = append(insertDocs, doc)
+		}
+	}
+
+	const numDocuments = 1000
+	const numCollections = 4
+
+	collections := [numCollections]*mongo.Collection{}
+
+	// batchMap is a mapping of collections to a list where each element is a slice
+	// of documents pre-batched at different batch sizes, it is safe for concurrent use by multiple goroutines
+	// as no data is modified concurrently.
+	//
+	//nolint:vet // we don't care about alignment there
+	type batchMap struct {
+		m          map[*mongo.Collection]*list.List
+		batchSizes []int
```

Comment:
I don't understand how we are creating different batchSizes, but when we call `InsertMany`, we are not benchmarking for different batchSizes :thinking: